### ISSUE TITLE
Behat WrongPathsException

### ISSUE THREAD

**Issue by jose-oliveira** [acquia/blt#2874]
My system information:
* Operating system type: MacO
* Operating system version: Sierra
* BLT version: BLT 9.1.0-alpha1

Output of `blt doctor`:
```
+---------------------------+------------------------------------------------------+
| Property                  | Value                                                |
+---------------------------+------------------------------------------------------+
| %paths.%root              | /var/www/shpoc/docroot                               |
| %paths.%site              | sites/default                                        |
| %paths.%modules           | sites/all/modules                                    |
| %paths.%themes            | sites/all/themes                                     |
| %paths.%config-sync       | /var/www/shpoc/config/default                        |
| %paths.%files             | sites/default/files                                  |
| %paths.%temp              | /tmp                                                 |
| %paths.%private           | /var/www/shpoc/files-private                         |
| admin-theme               | seven                                                |
| alias-searchpaths.0       | /var/www/shpoc/drush/sites                           |
| blt-version               | 9.1.0-alpha1                                         |
| bootstrap                 | Successful                                           |
| composer-version          | Composer version 1.6.5 2018-05-04 11:44:59           |
| config-sync               | /var/www/shpoc/config/default                        |
| db-driver                 | mysql                                                |
| db-hostname               | localhost                                            |
| db-name                   | drupal                                               |
| db-password               | drupal                                               |
| db-port                   | 3306                                                 |
| db-status                 | Connected                                            |
| db-username               | drupal                                               |
| drupal-settings-file      | sites/default/settings.php                           |
| drupal-version            | 8.5.4                                                |
| drush-alias-files.0       | /var/www/shpoc/drush/sites/example.acsf.site.yml     |
| drush-alias-files.1       | /var/www/shpoc/drush/sites/example.local.site.yml    |
| drush-alias-files.2       | /var/www/shpoc/drush/sites/legacy.site.yml           |
| drush-alias-files.3       | /var/www/shpoc/drush/sites/shpoc.site.yml            |
| drush-alias-files.4       | /var/www/shpoc/drush/sites/stubhubcms.site.yml       |
| drush-conf.0              | /var/www/shpoc/vendor/drush/drush/drush.yml          |
| drush-conf.1              | /var/www/shpoc/drush/drush.yml                       |
| drush-conf.2              | /var/www/shpoc/docroot/sites/default/local.drush.yml |
| drush-script              | /var/www/shpoc/vendor/bin/drush                      |
| drush-temp                | /tmp                                                 |
| drush-version             | 9.4-dev-g2342ecf                                     |
| files                     | sites/default/files                                  |
| install-profile           | lightning                                            |
| modules                   | sites/all/modules                                    |
| php-bin                   | /usr/bin/php7.1                                      |
| php-conf.0                | /etc/php/7.1/cli/php.ini                             |
| php-os                    | Linux                                                |
| private                   | /var/www/shpoc/files-private                         |
| root                      | /var/www/shpoc/docroot                               |
| site                      | sites/default                                        |
| stacks.drupal-vm.inited   | true                                                 |
| stacks.dev-desktop.inited | false                                                |
| temp                      | /tmp                                                 |
| theme                     | bartik                                               |
| themes                    | sites/all/themes                                     |
| uri                       | http://local.shpoc                                   |
+---------------------------+------------------------------------------------------+
+--------------------------------------+-----------------------------------------------------------+
| Check                                | Problem                                                   |
+--------------------------------------+-----------------------------------------------------------+
| NodeCheck:checkNodeVersionFileExists | Neither .nvmrc nor .node-version file found in repo root. |
+--------------------------------------+-----------------------------------------------------------+
```

When I run this command:
```
blt tests:behat:run -vvv
```

I get the following output:
```
[debug] Verifying that MySQL is available...
[Robo\Common\ProcessExecutor] Running '/var/www/shpoc/vendor/bin/drush' @self sqlq "SHOW DATABASES" in /var/www/shpoc/docroot
information_schema
drupal
[debug] Verifying that Drupal is installed...
[Robo\Common\ProcessExecutor] Running '/var/www/shpoc/vendor/bin/drush' @self sqlq "SHOW TABLES LIKE 'config'" in /var/www/shpoc/docroot
config
[debug] Verifying that Drupal is installed...
[Robo\Common\ProcessExecutor] Running '/var/www/shpoc/vendor/bin/drush' @self sqlq "SHOW TABLES LIKE 'config'" in /var/www/shpoc/docroot
config
Configuration for behat:
+---------------------+-------------------------------------------+
| Property            | Value                                     |
+---------------------+-------------------------------------------+
| behat.config        | /var/www/shpoc/tests/behat/local.yml      |
| behat.profile       | local                                     |
| behat.selenium.port | 4444                                      |
| behat.selenium.url  | http://127.0.0.1:4444/wd/hub              |
| behat.chrome.port   | 9222                                      |
| behat.chrome.args   |                                           |
| behat.paths.0       | /var/www/shpoc/tests/behat                |
| behat.tags          | ~ajax&&~experimental&&~lightningextension |
| behat.extra         |                                           |
| behat.web-driver    | chrome                                    |
+---------------------+-------------------------------------------+
+-------------------------------------------------------------------------------------------+-----------------------------------------------+
| Property                                                                                  | Value                                         |
+-------------------------------------------------------------------------------------------+-----------------------------------------------+
| imports.0                                                                                 | behat.yml                                     |
| local.suites.default.paths.features                                                       | /var/www/shpoc                                |
| local.suites.default.contexts.0.Drupal\FeatureContext                                     |                                               |
| local.suites.default.contexts.0.parameters.environment.screenshot_dir                     | /var/www/shpoc/reports                        |
| local.suites.default.contexts.1                                                           | Drupal\DrupalExtension\Context\DrupalContext  |
| local.suites.default.contexts.2                                                           | Drupal\DrupalExtension\Context\MinkContext    |
| local.suites.default.contexts.3                                                           | Drupal\DrupalExtension\Context\MessageContext |
| local.suites.default.contexts.4                                                           | Drupal\DrupalExtension\Context\DrushContext   |
| local.suites.default.contexts.5                                                           | Drupal\DrupalExtension\Context\ConfigContext  |
| local.extensions.DMore\ChromeExtension\Behat\ServiceContainer\ChromeExtension             |                                               |
| local.extensions.Behat\MinkExtension.browser_name                                         | chrome                                        |
| local.extensions.Behat\MinkExtension.javascript_session                                   | default                                       |
| local.extensions.Behat\MinkExtension.base_url                                             | http://local.shpoc                            |
| local.extensions.Behat\MinkExtension.show_cmd                                             | open %s                                       |
| local.extensions.Behat\MinkExtension.selenium2.wd_host                                    | http://127.0.0.1:4444/wd/hub                  |
| local.extensions.Behat\MinkExtension.selenium2.browser                                    | chrome                                        |
| local.extensions.Behat\MinkExtension.sessions.default.chrome.api_url                      | http://localhost:9222                         |
| local.extensions.Drupal\DrupalExtension.drupal.drupal_root                                | /var/www/shpoc/docroot                        |
| local.extensions.Drupal\DrupalExtension.drush.alias                                       | @self                                         |
| default.suites.default.paths.0                                                            | %paths.base%/features                         |
| default.extensions.Behat\MinkExtension.default_session                                    | goutte                                        |
| default.extensions.Behat\MinkExtension.browser_name                                       | chrome                                        |
| default.extensions.Behat\MinkExtension.javascript_session                                 | default                                       |
| default.extensions.Behat\MinkExtension.goutte                                             |                                               |
| default.extensions.Behat\MinkExtension.files_path                                         | %paths.base%/media                            |
| default.extensions.Drupal\DrupalExtension.blackbox                                        |                                               |
| default.extensions.Drupal\DrupalExtension.api_driver                                      | drupal                                        |
| default.extensions.jarnaiz\JUnitFormatter\JUnitFormatterExtension.filename                | report.xml                                    |
| default.extensions.jarnaiz\JUnitFormatter\JUnitFormatterExtension.outputDir               | %paths.base%/build/tests                      |
| default.extensions.Bex\Behat\ScreenshotExtension.screenshot_taking_mode                   | failed_steps                                  |
| default.extensions.Bex\Behat\ScreenshotExtension.image_drivers.local.screenshot_directory | /tmp                                          |
+-------------------------------------------------------------------------------------------+-----------------------------------------------+
[Filesystem\FilesystemStack] mkdir ["/var/www/shpoc/reports"]
[info] Killing running google-chrome processes...
[info] Killing all processes on port '9222'...
[info] Launching headless chrome...
[Robo\Common\ProcessExecutor] Running 'google-chrome' --headless --disable-web-security --remote-debugging-port=9222  http://localhost in /var/www/shpoc
[info] Waiting for response from http://localhost:9222...
[debug] cURL error 7: Failed to connect to localhost port 9222: Connection refused (see http://curl.haxx.se/libcurl/c/libcurl-errors.html)
[info] Waiting for response from http://localhost:9222...
[Testing\Behat] Running behat  --format pretty /var/www/shpoc/tests/behat --colors --no-interaction --stop-on-failure --strict --config /var/www/shpoc/tests/behat/local.yml --profile local --tags '~ajax&&~experimental&&~lightningextension' -v
No scenarios
No steps
0m8.14s (20.18Mb)

In ExerciseController.php line 111:
                                                                                                                                             
  [Behat\Testwork\Tester\Exception\WrongPathsException]                                                                                      
  No specifications found at path(s) `/var/www/shpoc/tests/behat`. This might be because of incorrect paths configuration in your `suites`.  
                                                                                                                                             

Exception trace:
 Behat\Testwork\Tester\Cli\ExerciseController->execute() at /var/www/shpoc/vendor/behat/behat/src/Behat/Testwork/Cli/Command.php:63
 Behat\Testwork\Cli\Command->execute() at /var/www/shpoc/vendor/symfony/console/Command/Command.php:251
 Symfony\Component\Console\Command\Command->run() at /var/www/shpoc/vendor/symfony/console/Application.php:946
 Symfony\Component\Console\Application->doRunCommand() at /var/www/shpoc/vendor/symfony/console/Application.php:248
 Symfony\Component\Console\Application->doRun() at /var/www/shpoc/vendor/behat/behat/src/Behat/Testwork/Cli/Application.php:124
 Behat\Testwork\Cli\Application->doRun() at /var/www/shpoc/vendor/symfony/console/Application.php:148
 Symfony\Component\Console\Application->run() at /var/www/shpoc/vendor/behat/behat/bin/behat:34

behat [-s|--suite SUITE] [-f|--format FORMAT] [-o|--out OUT] [--format-settings FORMAT-SETTINGS] [--init] [--lang LANG] [--name NAME] [--tags TAGS] [--role ROLE] [--story-syntax] [-d|--definitions DEFINITIONS] [--snippets-for [SNIPPETS-FOR]] [--snippets-type SNIPPETS-TYPE] [--append-snippets] [--no-snippets] [--strict] [--order ORDER] [--rerun] [--stop-on-failure] [--dry-run] [--] [<paths>]

[Testing\Behat]  Exit code 1 
[info] Killing running google-chrome processes...
[info] Killing all processes on port '9222'...
[error]  Behat tests failed! 
11.369s total time elapsed.
```

And I expected this to happen: Command to execute successfully. There aren't any custom Behat tests setup currently.

**Comment by jose-oliveira** [comment]
I tried all the steps mentioned in this issue: https://github.com/acquia/blt/issues/1979
And also tried to reboot and reprovision the VM. No success.

On Travis CI, this command is also failing with a different message:

```
> tests:behat:run
[warning] project.local.uri in blt.yml does not match local.extensions.Behat\MinkExtension.base_url in local.yml.
[warning] project.local.uri = http://127.0.0.1:8888
[warning] local.extensions.Behat\MinkExtension.base_url = http://local.shpoc
[warning] Behat is not configured properly.
BLT can (re)generate tests/behat/local.yml using tests/behat/example.local.yml.
[warning] project.local.uri in blt.yml does not match local.extensions.Behat\MinkExtension.base_url in local.yml.
[warning] project.local.uri = http://127.0.0.1:8888
[warning] local.extensions.Behat\MinkExtension.base_url = http://local.shpoc
[error]  Behat is not configured properly. Please run `blt doctor` to diagnose the issue. 
[error]  Command `tests:behat:run ` exited with code 1. 
```

**Comment by jose-oliveira** [comment]
So, after changing some fields on example.local.yml to match what Travis CI expects (probably a hacky solution), I'm seeing the same error on both CI and local.

**Comment by mikemadison13** [comment]
Have you tried running ```blt tests:behat:init:config ``` just to make sure that the config has been updated properly?

**Comment by jose-oliveira** [comment]
@mikemadison13 Yup, tried that and then rebooted and then reprovision the VM, no luck.

**Comment by jose-oliveira** [comment]
Ok, finally got this working on both local and CI:

**Local:** Behat fails is there aren't any tests to be run, so I added a tests/behat/features/Examples.feature from another project (running the 9.x-dev version) and that fixed it locally. Maybe this file should be generated with the ```blt tests:behat:init:config``` command?

**CI:** The file tests/behat/example.local.yml is generated with hardcoded values when running ```blt tests:behat:init:config``` on 9.1.0-alpha. I copied the same file from the 9.x-dev version and it uses variables instead. Then the tests run ok on Travis CI.
